parser/src/powdr.lalrpop

```diff
@@ -555,58 +555,72 @@ ProductOp: BinaryOperator = {
     "%" => BinaryOperator::Mod,
 }
 
-Power: Box<Expression> = {
-    <start:@L> <left:Term> <op:PowOp> <right:Power> <end:@R> => ctx.to_expr_with_source_ref(BinaryOperation { left, op, right }, start, end),
-    Unary,
+Power<StructOption>: Box<Expression> = {
+    <start:@L> <left:Term<StructOption>> <op:PowOp> <right:Power<StructOption>> <end:@R> => ctx.to_expr_with_source_ref(BinaryOperation { left, op, right }, start, end),
+    Unary<StructOption>,
 }
 
 PowOp: BinaryOperator = {
     "**" => BinaryOperator::Pow,
 }
 
-Unary: Box<Expression> = {
-    <start:@L> <op:PrefixUnaryOp> <expr:PostfixUnary> <end:@R> => ctx.to_expr_with_source_ref(UnaryOperation{op, expr}, start, end),
-    PostfixUnary,
+Unary<StructOption>: Box<Expression> = {
+    <start:@L> <op:PrefixUnaryOp> <expr:PostfixUnary<StructOption>> <end:@R> => ctx.to_expr_with_source_ref(UnaryOperation{op, expr}, start, end),
+    PostfixUnary<StructOption>,
 }
 
 PrefixUnaryOp: UnaryOperator = {
     "-" => UnaryOperator::Minus,
     "!" => UnaryOperator::LogicalNot,
 }
 
-PostfixUnary: Box<Expression> = {
-    <start:@L> <t:Term> <op:PostfixUnaryOp> <end:@R> => ctx.to_expr_with_source_ref(UnaryOperation{op, expr: t}, start, end),
-    Term,
+PostfixUnary<StructOption>: Box<Expression> = {
+    <start:@L> <t:Term<StructOption>> <op:PostfixUnaryOp> <end:@R> => ctx.to_expr_with_source_ref(UnaryOperation{op, expr: t}, start, end),
+    Term<StructOption>,
 }
 
 PostfixUnaryOp: UnaryOperator = {
     "'" => UnaryOperator::Next,
 }
 
-Term: Box<Expression> = {
-    <start:@L> <i:IndexAccess> <end:@R> => ctx.to_expr_with_source_ref(i, start, end),
-    <start:@L> <f:FunctionCall> <end:@R> => ctx.to_expr_with_source_ref(f, start, end),
+Term<StructOption>: Box<Expression> = {
+    StructOption,
+}
+
+TermWithStruct: Box<Expression> = {
+    TermCommon<TermWithStruct>,
+    //StructExpression<TermWithStruct>,
```

Comment:
In order to test it out, you could already parse struct expressions and then just turn them into something like an empty tuple value.